### DISCUSSION TITLE
how to configure @babel/preset-typescript with typescript-react code

### DISCUSSION THREAD

**Discussion by ooooevan** [babel/babel#13447] [Q&A] [answered]
I was writing typescript-react project（create-react-app） and it works fine.

now, i want to analysis my code with babel directly，but i get parsing error somethings

most of the codes are fine, just a few of strange code like `interface<> { name:string }` and `isNaN(<number>val)`, it works in the projects

My configuration is as follows
```
const babel = require("@babel/core");
babel.parseSync(codestring, {
    presets: [
      "@babel/preset-env",
      "@babel/preset-react",
      ["@babel/preset-typescript", { isTSX: true, allExtensions: true }]
    ],
  })
```

how should i change the configuration ? 
thank you

codesandbox: https://codesandbox.io/s/kind-cdn-xfg2w

**Comment by nicolo-ribaudo** [comment, marked as answer]
`isNaN(<number>val)` is not valid in TSX files, so you shouldn't tell Babel that all your files are TSX (with the `allExtensions` and `isTSX` options).

**Reply by ooooevan** [reply]
thank you，`isNaN(<number>val)` works when isTSX is `false`, but `interface<> { name:string }` still not working
